jaxlib/rocsolver.py

```diff
@@ -0,0 +1,304 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import operator
+
+import numpy as np
+
+from jaxlib import xla_client
+
+try:
+  from jaxlib import rocblas_kernels
+  for _name, _value in rocblas_kernels.registrations().items():
+    xla_client.register_custom_call_target(_name, _value, platform="ROCM")
+except ImportError:
+  pass
+
+# we have a single module for both rocsolver and rocblas functions
+rocsolver_kernels = rocblas_kernels
+
+
+_ops = xla_client.ops
+_Shape = xla_client.Shape
+
+# TODO(phawkins): remove after we no longer need to support old jax releases.
+def _unpack_builder(c):
```

Comment:
You don't need this compatibility code, since this module never existed in ancient `jax` versions.

Reply:
https://github.com/google/jax/pull/5115/commits/e51a33c2b7d3dff8bab8baa208d8c1178728f274